sklearn/cluster/_birch.py

```diff
@@ -745,4 +749,4 @@ def _global_clustering(self, X=None):
             self.subcluster_labels_ = clusterer.fit_predict(self.subcluster_centers_)
 
         if compute_labels:
-            self.labels_ = self.predict(X)
+            self.labels_ = self._predict(X)
```

Comment:
wouldn't a cleaner API be like `self.predict(X, validate_input=False)`, or `self.validate_input(predict=False).predict(X)`?

Reply:
> self.predict(X, validate_input=False)

I don't think introducing a public arg for that is cleaner. I find it clean that we use a private function internally and expose a public function that does extra validation.

> self.validate_input(predict=False).predict(X)

I don't get the `predict=False` arg, could explain more what you have in mind ?

Reply:
There are other reasons why a public, or a developer API, would be nice to have when it comes to [skipping] validation: https://github.com/scikit-learn/scikit-learn/issues/16653#issuecomment-948058660

The `predict=False` would kinda set a flag in the estimator to skip the validation in a certain method.

Reply:
> The predict=False would kinda set a flag in the estimator to skip the validation in a certain method.

I think adding more state to the estimator after `__init__` is outside the scope of this PR, but we can use this PR as a motivation to do it. It would kind of be like "inference mode".

> self.predict(X, validate_input=False)

I think it would be very nice to have this type of kwarg everywhere. It would be similar to the `check_finite` flag in SciPy. (Every year I see the "Scikit-learn is slow during prediction" and it comes down to the validation we do.)

In both cases, I do not think we should change public API with a bug fix PR.